### PR TITLE
Cyclist improvements

### PR DESCRIPTION
**Pull request by Slotos**

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -3,7 +3,9 @@ source 'https://rubygems.org'
 # Specify your gem's dependencies in tsv.gemspec
 gemspec
 
-gem "codeclimate-test-reporter", group: :test, require: nil
-gem "rake"
-gem "rspec"
-gem "pry"
+group "test" do
```
Comment (brain-geek) [inline Gemfile]: This gemfile is only used in running tests. In real gem usage, gemspec is used.

**Comment by Slotos** [comment]
Yeah, I'm leaning towards ditching `Cyclist` inheritance idea. The initial idea was to provide easy way to extend the functionality, but it doesn't seem to be solving a real world problem. Incidentally it's possible to imitate ruby core CSV naming, since it solves almost the same problem.

`#with_header` and `#withour_header` can be simplified, but removing them doesn't change neither operational nor code complexity. As such I'd prefer leaving them in place.

I'll get on with cutting out Cyclist inheritance then. One elephant at a time.

File: Gemfile (hunk 1 above)
Comment (Slotos) [inline Gemfile]: I had a pause remembering how it's used. Explicit grouping doesn't change anything functionally, but adds visual reminder.

**Comment by Slotos** [comment]
Unexpected bonus from this refactor: I could ditch the test that needed five lines of comments to make sense.

**Comment by Slotos** [comment]
A bit of a late consideration. `TSV::Table` is also a misleading name, since we don't really provide a table, but an Enumerator wrapper. Though its intent is still clearer.

**Comment by brain-geek** [comment]
I love those new improvements.

If we won't have anymore ideas/PRs then I'll release this version on Rubygems in a few days.

**Pull request merged**